src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -91,57 +93,62 @@ internal static ProcessModule GetMainModule(Process targetProcess)
         /// <summary>
         /// Retrieve the parent process of a process.
         /// 
-        /// This is an extremely expensive operation, as WMI
-        /// needs to work with an ugly Win32 API. The Win32 API
-        /// creates a snapshot of every process in the system, which
-        /// you then need to iterate through to find your process and
-        /// its parent PID.
-        ///
-        /// Also, since this is PID based, this API is only reliable
-        /// when the process has not yet exited.
+        /// Previously this code used WMI, but WMI is causing a CPU spike whenever the query gets called as it results in 
+        /// tzres.dll and tzres.mui.dll being loaded into every process to conver the time information to local format.
+        /// For perf reasons, we result to P/Invoke.
         /// </summary>
         ///
         /// <param name="current">The process we want to find the
         /// parent of</param>
         internal static Process GetParentProcess(Process current)
         {
-            string wmiQuery = String.Format(CultureInfo.CurrentCulture,
-                                            "Select * From Win32_Process Where Handle='{0}'",
-                                            current.Id);
-
-            using (CimSession cimSession = CimSession.Create(null))
-            {
-                IEnumerable<CimInstance> processCollection =
-                    cimSession.QueryInstances("root/cimv2", "WQL", wmiQuery);
-
-                int parentPid =
-                    processCollection.Select(
-                        cimProcess =>
-                        Convert.ToInt32(cimProcess.CimInstanceProperties["ParentProcessId"].Value,
-                                        CultureInfo.CurrentCulture)).FirstOrDefault();
+            int parentPid = 0;
```

Comment:
Do we have any Pester tests that cover this code change? If not, please create a new test or port some internal tests to the new environment.